sympy/sets/tests/test_sets.py

```diff
@@ -941,3 +942,13 @@ def test_issue_10113():
 def test_issue_10248():
     assert list(Intersection(S.Reals, FiniteSet(x))) == [
         And(x < oo, x > -oo)]
+
+
+def test_set_in_interval():
+    x = Symbol('x', real=True)
+    assert x in Interval(x, x+4)
+    assert Interval(1, 2) not in Interval(0, 5)
```

Comment:
@smichr I raised a topic along this line [here](https://github.com/sympy/sympy/issues/7805) before opening a PR.
Even I am of the mind that atleast a FiniteSet can be `in` an Interval.